clang/include/clang/Basic/DiagnosticGroups.td

```diff
@@ -1450,7 +1450,6 @@ def MicrosoftAnonTag : DiagGroup<"microsoft-anon-tag">;
 def MicrosoftCommentPaste : DiagGroup<"microsoft-comment-paste">;
 def MicrosoftEndOfFile : DiagGroup<"microsoft-end-of-file">;
 def MicrosoftInaccessibleBase : DiagGroup<"microsoft-inaccessible-base">;
-def MicrosoftStaticAssert : DiagGroup<"microsoft-static-assert">;
```

Comment:
I was curious whether we should leave the diagnostic group so it's not disruptive, but: https://sourcegraph.com/search?q=context:global+Wno-microsoft-static-assert&patternType=keyword&sm=0

Given there's so little use in the wild, I think it's better to just pull the group entirely so we don't keep it around as tech debt.

Reply:
Might as well! Better to over-communicate than under.

Reply:
Added.